### PR TITLE
Avoid Dynamic Web project

### PR DESCRIPTION
**Pull request by chanseokoh**
Part of #2909.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/ToServlet25SourceQuickFixTest.java [hunk 1]
```diff
@@ -68,12 +58,10 @@ public void testConvertServlet() throws CoreException {
     IDocument document = viewer.getDocument();
     String contents = document.get();
     assertFalse(contents.contains("version='3.1'"));
+    assertTrue(contents.contains("version=\"2.5\""));
 
     // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1527
     editorPart.doSave(new NullProgressMonitor());
-
-    ProjectUtils.waitForProjects(project);
-    assertEquals(0, file.findMarkers(SERVLET_MARKER, true, IResource.DEPTH_ZERO).length);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/ToServlet25SourceQuickFixTest.java]: The justification of removing this code being that it's an end-to-end test, and this outcome is already tested elsewhere.  Right?  Seems reasonable, just want to verify.

File: plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorPluginTest.java [hunk 2]
```diff
@@ -48,10 +46,12 @@
   private IJavaProject javaProject;
   private IResource resource;
   @Rule public TestProjectCreator projectCreator = new TestProjectCreator().withFacets(
-      JavaFacet.VERSION_1_7, WebFacetUtils.WEB_25, AppEngineStandardFacet.JRE7);
+      JavaFacet.VERSION_1_7);
 
-  @Before
-  public void setUp() throws CoreException {
+  private void setUpProject(boolean dynamicWebProject) throws CoreException {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorPluginTest.java]: Perhaps we'd be better separating the tests into two classes, one with tests requiring only a plain project, the other for tests requiring the DWP.

File: plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/ToServlet25SourceQuickFixTest.java (hunk 1 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/ToServlet25SourceQuickFixTest.java]: This is not tested elsewhere actually, but I did want to downgrade this end-to-end test to a smaller unit test. I think it's a reasonable change considering the new assertion.

File: plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorPluginTest.java (hunk 2 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorPluginTest.java]: Let me try that.

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorPluginTest.java]: Done!

**Pull request merged**